### ISSUE TITLE
cloudflare SSL error: tlsv1 alert internal error

### ISSUE THREAD

**Issue by centminmod** [eribertomota/axel#35]
seems axel 2.11 doesn't like cloudflare hosted HTTPS sites with ECDSA ciphers 

```
axel --verbose https://centmin.sh/centminmodparts/ccache/ccache-3.3.2.tar.gz
Initializing download: https://centmin.sh/centminmodparts/ccache/ccache-3.3.2.tar.gz
SSL error: tlsv1 alert internal error
```

wget works fine

```
wget -O /dev/null https://centmin.sh/centminmodparts/ccache/ccache-3.3.2.tar.gz
--2016-10-04 21:03:18--  https://centmin.sh/centminmodparts/ccache/ccache-3.3.2.tar.gz
Resolving centmin.sh (centmin.sh)... 104.27.173.154, 104.27.172.154
Connecting to centmin.sh (centmin.sh)|104.27.173.154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 445945 (435K) [application/octet-stream]
Saving to: ‘/dev/null’

/dev/null                                                      100%[==================================================================================================================================================>] 435.49K   688KB/s    in 0.6s    

2016-10-04 21:03:20 (688 KB/s) - ‘/dev/null’ saved [445945/445945]
```

wget version

```
wget -V
GNU Wget 1.18 built on linux-gnu.

-cares +digest -gpgme +https +ipv6 +iri +large-file -metalink +nls 
+ntlm +opie -psl +ssl/openssl 

Wgetrc: 
    /usr/local/etc/wgetrc (system)
Locale: 
    /usr/local/share/locale 
Compile: 
    gcc -DHAVE_CONFIG_H -DSYSTEM_WGETRC="/usr/local/etc/wgetrc" 
    -DLOCALEDIR="/usr/local/share/locale" -I. -I../lib -I../lib -I 
    /usr/local/include -DHAVE_LIBSSL -DNDEBUG -O2 -g -pipe -Wall 
    -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong 
    --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic 
Link: 
    gcc -I /usr/local/include -DHAVE_LIBSSL -DNDEBUG -O2 -g -pipe -Wall 
    -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong 
    --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic 
    -L /usr/local/lib -lpcre -lssl -lcrypto -lz -lidn ftp-opie.o 
    openssl.o http-ntlm.o ../lib/libgnu.a 
```

**Comment by cathalgarvey** [comment]
It is [alleged here](https://askubuntu.com/questions/214018/how-to-make-wget-faster-or-multithreading) that Axel fudges SSL/TLS and uses plain HTTP to download. If so, this would explain why even slightly secure servers refuse to allow Axel to finish connecting.

If true, it's a complete shambles and needs to be fixed ASAP.

**Comment by sdt** [comment]
Sorry for the late response there @centminmod 

There is a patch which _may_ fix your issue here: https://github.com/sdt/axel/tree/issue/29 . I haven't had time to look at this lately.

@cathalgarvey - that used to be the case. Actual HTTPS handling got added in e1ebab044c0476c09faadbb88a99ea4c839bf19c. There are still some outstanding issues to be fixed. Pull requests are most welcome.

**Comment by ismaell** [comment]
Fixed by commit 896310575868ec883a30f768c32771a5bf6dc451.
